ipatests/test_integration/test_trust.py

```diff
@@ -344,3 +355,121 @@ def test_trustdomains_found_in_external_rootdomain_trust(self):
 
     def test_remove_external_rootdomain_trust(self):
         self.remove_trust(self.ad)
+
+    # Test for one-way forest trust with shared secret
+
+    def test_establish_forest_trust_with_shared_secret(self):
+        self.configure_dns_and_time(self.ad)
+        tasks.configure_windows_dns_for_trust(self.ad, self.master)
+
+        # this is a workaround for
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1711958
+        self.master.run_command(
+            ['ipa', 'dnsrecord-add', self.master.domain.name,
+             self.srv_gc_record_name,
+             '--srv-rec', self.srv_gc_record_value])
+
+        # create windows side of trust using powershell bindings
+        # to .Net functions
+        ps_cmd = (
+            '[System.DirectoryServices.ActiveDirectory.Forest]'
+            '::getCurrentForest()'
+            '.CreateLocalSideOfTrustRelationship("{}", 1, "{}")'.format(
+                self.master.domain.name, self.shared_secret))
+        self.ad.run_command(['powershell', '-c', ps_cmd])
+
+        # create ipa side of trust
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain, shared_secret=self.shared_secret)
+
+    def test_trustdomains_found_in_forest_trust_with_shared_secret(self):
+        result = self.master.run_command(
+            ['ipa', 'trust-fetch-domains', self.ad.domain.name],
+            raiseonerr=False)
+        assert result.returncode == 1
+        self.check_trustdomains(
+            self.ad_domain, [self.ad_domain, self.ad_subdomain])
+
+    def test_user_gid_uid_resolution_in_forest_trust_with_shared_secret(self):
+        """Check that user has SID-generated UID"""
+        # Using domain name since it is lowercased realm name for AD domains
+        testuser = 'testuser@%s' % self.ad_domain
+        result = self.master.run_command(['getent', 'passwd', testuser])
+
+        # This regex checks that Test User does not have UID 10042 nor belongs
+        # to the group with GID 10047
+        testuser_regex = r"^testuser@%s:\*:(?!10042)(\d+):(?!10047)(\d+):"\
+                         r"Test User:/home/%s/testuser:/bin/sh$"\
+                         % (re.escape(self.ad_domain),
+                            re.escape(self.ad_domain))
+
+        assert re.search(
+            testuser_regex, result.stdout_text), result.stdout_text
```

Comment:
i think 
assert re.search( testuser_regex, result.stdout_text) is sufficient and another result.stdout_text not needed..

Reply:
result.stdout_text after comma will be displayed when asserttion fails. This makes debugging easier, as you see output in test logs. Otherwise we would need to manually reproduce test scenario to find out what went wrong